src/plugins/data/public/index.ts

```diff
@@ -422,6 +422,7 @@ export {
   SearchBarProps,
   StatefulSearchBarProps,
   IndexPatternSelectProps,
+  IndexPatternSelectPublicProps,
```

Comment:
I would suggest exporting `IndexPatternSelectPublicProps as IndexPatternSelectProps` to have more consistent naming.
Anyway I don't think we need to export both.

Reply:
To avoid confusion with `IndexPatternSelectProps` meaning 2 things, I replaced `IndexPatternSelectPublicProps` with `IndexPatternSelectProps` and then created `IndexPatternSelectInternalProps`. `IndexPatternSelectInternalProps` is the union of `IndexPatternSelectProps` and properties provided by data.